core/src/main/java/org/jruby/RubyThread.java

```diff
@@ -604,44 +612,27 @@ private IRubyObject startThread(ThreadContext context, Runnable runnable) throws
-        final String file, final int line, final boolean newThread) {
+    private static void initThreadName(final Ruby runtime, final Thread thread, final String file, final int line) {
         // "Ruby-0-Thread-16: (irb):21"
-        // "Ruby-0-Thread-17@worker#1: (irb):21"
         final String newName;
-        final String setName = getNameOrNull();
-        final String currentName = thread.getName();
-        if ( currentName != null && currentName.startsWith(RUBY_THREAD_PREFIX) ) {
-            final int i = currentName.indexOf('@'); // Thread#name separator
-            if ( i == -1 ) { // name not set yet: "Ruby-0-Thread-42: FILE:LINE"
-                int end = currentName.indexOf(':');
-                if ( end == -1 ) end = currentName.length();
-                final String prefix = currentName.substring(0, end);
-                newName = currentName.replace(prefix, prefix + '@' + setName);
-
-            }
-            else { // name previously set: "Ruby-0-Thread-42@foo: FILE:LINE"
-                final String prefix = currentName.substring(0, i); // Ruby-0-Thread-42
-                int end = currentName.indexOf(':', i);
-                if ( end == -1 ) end = currentName.length();
-                final String prefixWithName = currentName.substring(0, end); // Ruby-0-Thread-42@foo:
-                newName = currentName.replace(prefixWithName, setName == null ? prefix : (prefix + '@' + setName));
-            }
-        }
-        else if ( newThread ) {
-            final StringBuilder name = new StringBuilder(24);
-            name.append(RUBY_THREAD_PREFIX).append(runtime.getRuntimeNumber());
-            name.append('-').append("Thread-").append(incAndGetThreadCount(runtime));
-            if ( setName != null ) name.append('@').append(setName);
-            if ( file != null ) { // in JIT we seem to get "" as file and line 0
-                name.append(':').append(' ').append(file).append(':').append(line + 1);
-            }
-            newName = name.toString();
-        }
-        else return; // not a new-thread that and does not match out Ruby- prefix
-        // ... very likely user-code set the java thread name - thus do not mess!
-        try { thread.setName(newName); }
-        catch (SecurityException ignore) { } // current thread can not modify
+        final StringBuilder name = new StringBuilder(24);
+        name
+                .append(RUBY_THREAD_PREFIX)
+                .append(runtime.getRuntimeNumber())
+                .append('-')
+                .append("Thread-")
+                .append(incAndGetThreadCount(runtime));
+        if ( file != null ) {
+            name
+                    .append(':')
+                    .append(' ')
+                    .append(file)
+                    .append(':')
+                    .append(line + 1);
+        }
+        newName = name.toString();
+
+        thread.setName(newName);
```

Comment:
should `catch (SecurityException ignore) { } // current thread can not modify` like before (an bellow)